### ISSUE TITLE
Add processes inventory for Mac OS X

### ISSUE THREAD

**Issue by Skeptor** [wazuh/wazuh#3295]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.11.0 | Syscollector | Agent | Packages/Sources | Mac OS X |

This issue is derivated from issue #2923 

The scan gets a list of current pids on the system and then it iterates through the list of pids to obtain the information of each process.

The first code version of the scan is:
```
pid_t * pids = calloc(0x1000, 1);
	struct proc_taskallinfo task_info;
  register struct passwd *pw;
  int priority;
  int count = proc_listallpids(pids, 0x1000);
	int error;

    printf("count=%u\n", count) ;

    for( int index=0; index < count; ++index) {
		pid_t pid = pids[ index ] ;
		printf("\n	PID %u:\n    ", pid);

		int st = proc_pidinfo(pid, PROC_PIDTASKALLINFO, 0,
							&task_info, PROC_PIDTASKALLINFO_SIZE);

		if (st != PROC_PIDTASKALLINFO_SIZE) {
			fprintf(stderr, "Cannot get process info\n");
			continue;
		}

		printf("Father's pid: %d | ", task_info.pbsd.pbi_ppid);
		printf("Group's pid: %d | ", task_info.pbsd.pbi_gid);
		printf("name: %s | ",      task_info.pbsd.pbi_name);
		
		pw = getpwuid((int)task_info.pbsd.pbi_uid);
		if(pw)
			printf("user_name: %s | ", pw->pw_name);
		
		printf("nice: %d | ", task_info.pbsd.pbi_nice);
		printf("status: %d | ", task_info.pbsd.pbi_status);
		printf("priority: %d | ", task_info.ptinfo.pti_priority);

		printf("\n");
    }
```

With an output like:

```
PID 50:
    Father's pid: 1 | Group's pid: 0 | name: syslogd | user_name: root | nice: 0 | status: 2 | priority: 4 | 

	PID 1:
    Father's pid: 0 | Group's pid: 0 | name: launchd | user_name: root | nice: 0 | status: 2 | priority: 31 | 

	PID 0:
    Father's pid: 0 | Group's pid: 0 | name: kernel_task | user_name: root | nice: 0 | status: 2 | priority: 81 | 
```

Remaining tasks
--------

- [x] Convert the field status from integer to text
- [x] Add the scan to the syscollector module of BSD.
- [x] Check the compilation in BSD systems. It is necessary to check if it works for the rest of BSD platforms or it has to be isolated only for Mac OS X
- [x] Check Memory status. Run valgrind on `wazuh-modulesd`

**Comment by Skeptor** [comment]
The conversion for the field status is in the library `sys/proc.h`:
```
/* stat codes */
#define	SSLEEP	1		/* awaiting an event */
#define	SWAIT	2		/* (abandoned state) */
#define	SRUN	3		/* running */
#define	SIDL	4		/* intermediate state in process creation */
#define	SZOMB	5		/* intermediate state in process termination */
#define	SSTOP	6		/* process being traced */
```

I've changed the scan to add the conversion as: `SLEEP, WAIT, RUNNING, IDLE, ZOMBIE, STOPPED` but those names are just for developing purpose as the process inventory needs to have descriptive names.

The next step is the integration of the scan in the syscollector module to be able to send the inventory to the manager.

**Comment by Skeptor** [comment]
I'm working on the branch: https://github.com/wazuh/wazuh/compare/issue-3295

The commit https://github.com/wazuh/wazuh/commit/796f775f90a3c1b50dadeab9c19c09cce6c21649 adds the process scan to the Mac OS agents.
The next step is to ensure that the output is correct in different situations, change the state of the process to return descriptive messages and check the memory use of `modulesd`

**Comment by Skeptor** [comment]
The commit https://github.com/wazuh/wazuh/commit/014c7b58362e2db8cb0839d854c201f87f7c6ccb remove the ignore of process scan for MAC agents and change some fields of the scan to return the same information than the Linux scan.

Also the function is defined only for Mac OS because I tried to compile the code in FreeBSD but the struct `proc_taskallinfo`  wasn't defined.

**Comment by Skeptor** [comment]
I've found the updated list of states and I've updated the first comment of this issue to use the new list as reference.

Now I got this results in Kibana
![image](https://user-images.githubusercontent.com/9034923/57637491-db437780-75ab-11e9-9eca-337f11430368.png)

The last check of the issue is the memory status but I'm facing troubles to run `valgrind` on Mac OS Sierra because I got the following error:
`2019/05/13 18:21:58 wazuh-modulesd[83731] main.c:80 at main(): CRITICAL: (1130): Unable to switch to group 'ossec' due to [(78)-(Function not implemented)].`

**Comment by chemamartinez** [comment]
Merged #3322. GJ!
